contracts/tests/EspressoMigrationAction.t.sol

```diff
@@ -0,0 +1,236 @@
+                nativeToken: address(0),
+                deployFactoriesToL2: true,
+                maxFeePerGasForRetryables: MAX_FEE_PER_GAS,
+                batchPosterManager: batchPosterManager
+            });
+        rollupAddress = rollupCreator.createRollup{value: factoryDeploymentFunds}(
+            deployParams
+        );
+
+        vm.stopPrank();
+    }
+
+    function _prepareRollupDeployment()
+        internal
+        returns (
+            IOneStepProofEntry ospEntry,
+            IChallengeManager challengeManager,
+            IRollupAdmin rollupAdminLogic,
+            IRollupUser rollupUserLogic
+        )
+    {
+        //// deploy challenge stuff
+        ospEntry = new OneStepProofEntry(
+            new OneStepProver0(),
+            new OneStepProverMemory(),
+            new OneStepProverMath(),
+            new OneStepProverHostIo()
+        );
+        challengeManager = new ChallengeManager();
+
+        //// deploy rollup logic
+        rollupAdminLogic = IRollupAdmin(new RollupAdminLogic());
+        rollupUserLogic = IRollupUser(new RollupUserLogic());
+
+        return (ospEntry, challengeManager, rollupAdminLogic, rollupUserLogic);
+    }
+
+    function _getProxyAdmin(address proxy) internal view returns (address) {
+        bytes32 adminSlot = bytes32(uint256(keccak256("eip1967.proxy.admin")) - 1);
+        return address(uint160(uint256(vm.load(proxy, adminSlot))));
+    }
+
+    function test_migrateToEspresso() public{
+        //begin by seting pre-requisites in the vm so the test can get the data it needs.
+        IRollupCore rollup = IRollupCore(rollupAddress);
+       
+        address upgradeExecutorExpectedAddress = computeCreateAddress(address(rollupCreator), 4);
+        //ensure we have the correct address for the proxy admin
+        assertEq(
+            ProxyAdmin(_getProxyAdmin(address(rollup.sequencerInbox()))).owner(),
+            upgradeExecutorExpectedAddress,
+            "Invalid proxyAdmin's owner"
+        );
+
+        IUpgradeExecutor _upgradeExecutor = IUpgradeExecutor(
+            upgradeExecutorExpectedAddress
+        );
+
+
+        vm.setEnv("NEW_OSP_ENTRY", Strings.toHexString(uint256(uint160(address(newOspEntry)))));
+        vm.setEnv("CURRENT_OSP_ENTRY", Strings.toHexString(uint256(uint160(address(originalOspEntry)))));
+        vm.setEnv("ROLLUP_ADDRESS", Strings.toHexString(uint256(uint160(address(rollupAddress)))));
+        vm.setEnv("PROXY_ADMIN", Strings.toHexString(uint256(uint160(computeCreateAddress(address(rollupCreator), 1)))));
+        vm.setEnv("NEW_WASM_MODULE_ROOT", Strings.toHexString(uint256(keccak256("newRoot"))));
+        vm.setEnv("CURRENT_WASM_MODULE_ROOT", Strings.toHexString(uint256(keccak256("wasm"))));
+        bytes memory data = abi.encodeWithSelector(
+            OspMigrationAction.perform.selector  
+        );
+        
+        address migration = address(new EspressoOspMigrationAction());
+
+        vm.prank(rollupOwner);
```

Comment:
consider adding more tests that check for negative scenarios e.g. expecting a revert of a non-rollup owner tries to execute the upgrade